src/components/inputs/reactQueryBuilder/FieldSelector.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FullOption, toFlatOptionArray, ValueSelectorProps } from 'react-querybuilder';
+import { Autocomplete, TextField } from '@mui/material';
+
+function FieldSelector(props: Readonly<ValueSelectorProps>) {
+    const { options, className, value, disabled, handleOnChange } = props;
```

Comment:
```suggestion
function FieldSelector({ options, className, value, disabled, handleOnChange }: Readonly<ValueSelectorProps>) {
```
New cooler way to do the same thing. I think @TheMaskedTurtle prefer like this but not sure at all. So do as you wish.

Reply:
Agreed with Mathieu, now destructuring in the signature is in convention except some particular case.
In addition, I dont see the befifice of using Readonly<T> for props if we destruct in the signature